importlib_metadata/_functools.py

```diff
@@ -1,9 +1,19 @@
 import types
 import functools
 
+from typing import Callable, TypeVar
+
+
+CallableT = TypeVar("CallableT", bound=Callable[..., object])
+
 
 # from jaraco.functools 3.3
-def method_cache(method, cache_wrapper=None):
+def method_cache(
```

Comment:
I started looking into porting this functionality to jaraco.functools in jaraco/jaraco.functools#19. Unfortunately, introduction of the types triggers failures in pytest-mypy.

Reply:
it wasn't until I tried porting these changes back to jaraco.functools that I remembered that this function wasn't a direct copy from jaraco.functools, but a modified version with `_special_method_cache` removed (unused in this context), reminding me of the dangers of copy/pasting code. Fortunately, the tests were sufficient to catch the failure. Probably when bringing a type-annotated version of this function over here, we should be careful to stub `_special_method_cache` rather than modifing the function.